src/model/chunk.py

```diff
@@ -0,0 +1,36 @@
+## This file contains the function to chunk the input text into smaller pieces for the model to process
+## The input text is tokenized and then split into chunks of size chunk_size
+
+from transformers import AutoTokenizer
+import torch
+
+def chunk(inp: str, 
+          tokenizer: AutoTokenizer, 
+          chunk_size: int = 256, 
+          overlapping_len: int = 3,
+          num_chunks: int = 128) -> list:
+    tokenized_txt = tokenizer(inp, return_tensors="pt")["input_ids"]
+    token_len = tokenized_txt.size()[1]
+    
+    if token_len <= chunk_size:
+        return [tokenized_txt]
+    else:
+        chunks = []
+        chunks.append(tokenized_txt.view(-1)[:chunk_size])
+        for i in range(chunk_size, token_len, chunk_size):
+            if len(chunks) == num_chunks:
+                # only return num_chunks chunks so that it is not stuck on the same sample for too long
+                break
+            if i + chunk_size > token_len:
+                new_chunk = tokenized_txt.view(-1)[i-overlapping_len:]
+                new_chunk_len = new_chunk.size()[0]
+                new_chunk = new_chunk.view(1, new_chunk_len)
+                pad_len = chunk_size - new_chunk_len
+                padded_chunk = torch.cat((new_chunk, torch.full((1, pad_len), tokenizer.eos_token_id)), dim=1)
```

Comment:
would be better practice to pad with `tokenizer.pad_token_id` and in the top of the function check if the tokenizer has a `pad_token_id` set. if the tokenizer doesn't have a `pad_token_id` set then we can default to using `eos_token_id`, but if the tokenizer we are provided in this function already has a `pad_token_id` set we should be using that over `eos_token_id`

Reply:
Implemented usage of ```tokenizer.pad_token_id``` https://github.com/MichiganDataScienceTeam/F24-mini-copilot/pull/24/commits/926beedac4e3a431f36582d81dae250ddf5e172f
Did not implement checking of ```pad_token_id``` existence because it might be possible (and in that case maybe better) to do that in the retraining of the tokenizer